src/App.tsx

```diff
@@ -1,61 +1,110 @@
 import './App.scss';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const todos = todosFromServer.map((todo) => {
+  return {
+    ...todo,
+    user: usersFromServer.find(user => user.id === todo.userId) || null,
+  };
+});
 
 export const App = () => {
+  const [title, handleTitle] = useState('');
+  const [userName, handleOption] = useState('');
+  const [visibleTodos, setTodos] = useState(todos);
+  const [titleIsValid, setTitle] = useState(true);
+  const [userNameIsValid, setUsername] = useState(true);
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => {
+          event.preventDefault();
+          if (title && userName) {
+            const newTodoId = [...visibleTodos]
+              .sort((a, b) => a.id - b.id).reverse()[0].id;
+            const newUser = usersFromServer
+              .find(user => user.name === userName) || null;
+            const newTodo: Todo = {
+              id: newTodoId + 1,
+              title,
+              completed: false,
+              userId: newUser?.id || 0,
```

Comment:
userId can't be 0 because user has to select person in form

Reply:
agree